tests/test_views.py

```diff
@@ -86,3 +96,103 @@ def test_post_detail_failed(self):
+        self.assertTemplateUsed(http_response, 'blog/post_detail.html')
+        new_post = Post.objects.last()
+        self.assertRedirects(
+            http_response,
+            reverse(
+                'post_detail',
+                kwargs={'pk': new_post.pk},
+            ),
+        )
+        self.assertEqual(new_post.title, 'Test Title')
+        self.assertEqual(new_post.text, 'Test Text')
+        self.assertEqual(new_post.author, self.user)
+        self.assertIsNotNone(new_post.published_date)
+        self.assertEqual(Post.objects.count(), initial_post_count + 1)
+
+    def test_post_new_invalid_form_authorized(self):
+        """Test handling of invalid form submission for new post."""
+        self.client.force_login(self.user)
+        url = reverse('post_new')
+        http_response = self.client.post(url, self.invalid_form_post)
+        self.assertEqual(http_response.status_code, 200)
+        self.assertTemplateUsed(http_response, 'blog/post_edit.html')
+        self.assertContains(
+            http_response, 'This field is required.', html=True,
+        )
+
+    def test_post_edit_get_authorized(self):
+        """Ensure authorized users have access to edit the post."""
+        self.client.force_login(self.user)
+        post = self.current_post
+        url = reverse('post_edit', kwargs={'pk': post.pk})
+        http_response = self.client.get(url)
+        self.assertEqual(http_response.status_code, 200)
+        self.assertTemplateUsed(http_response, 'blog/post_edit.html')
+
+    def test_post_edit_get_unauthorized(self):
+        """Ensure unauthorized users are redirected when edit a post."""
+        self.client.logout()
+        url = reverse('post_edit', kwargs={'pk': self.current_post.pk})
+        http_response = self.client.get(url)
+        self.assertRedirects(http_response, f'/accounts/login/?next={url}')
+
+    def test_post_edit_valid_form_authorized(self):
+        """Check the correctness of updating the post after editing."""
+        self.client.force_login(self.user)
+        post = self.current_post
+        url = reverse('post_edit', kwargs={'pk': post.pk})
+        updated_form_data = self.valid_form_post.copy()
+        updated_form_data['title'] = 'Updated Title'
+        updated_form_data['text'] = 'Updated Text'
+        initial_published_date = post.published_date
+        http_response = self.client.post(url, updated_form_data, follow=True)
+        self.assertEqual(http_response.status_code, 200)
+        self.assertTemplateUsed(http_response, 'blog/post_detail.html')
+        self.assertRedirects(
+            http_response, reverse(
+                'post_detail', kwargs={'pk': post.pk},
+            ),
+        )
+        post.refresh_from_db()
+        self.assertEqual(post.title, 'Updated Title')
+        self.assertEqual(post.text, 'Updated Text')
+        self.assertEqual(post.author, self.user)
+        self.assertNotEqual(post.published_date, initial_published_date)
```

Comment:
While this is a useful check, the best way of dealing with the time changes is freezing it. There are libraries like freezegun and similar that monkey-patch the stdlib that could be used. But since you did it manually in the past already, I suggest repeating the same technique here: https://github.com/kpi-web-guild/django-girls-blog-OlenaEfymenko/blob/f6ccd09/tests/test_views.py#L60-L63.